Minesweeper/RandomMinePlacementGenerator.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+
+namespace Minesweeper
+{
+    public class RandomMinePlacementGenerator : IMinePlacementGenerator
+    {
+        private int _rows;
+        private int _cols;
+        private Random _rng;
+        public RandomMinePlacementGenerator(int rows, int cols)
+        {
+            if (rows < 0 || cols < 0)
+                throw new ArgumentException("Invalid argument.");
+            _rows = rows;
+            _cols = cols;
+            _rng = new Random();
+        }
+        public Tuple<int,int> Next()
+        {
+            return new Tuple<int,int>(_rng.Next(0, _rows), _rng.Next(0, _cols));
+        }
+    }
```

Comment:
I feel like the logic of this class should be inside Generate method of the BoardGenerator. But since it is in a separate class, shouldn't this be tested as well? I don't get why there are tests for Mod2MineGenerator which presumably isn't random(which is what we actually want) and there are no tests for this class. i think testing randomness is difficult and totally not the point of this task so I would skip this anyway, I am just confused by some tests and the design of this. 

Btw "throw new ArgumentException("Invalid argument.");" that isn't very helpful. I would maybe remove the message or at least make it more descriptive.